scripts/zkai_snarkjs_statement_envelope_benchmark.py

```
@@ -0,0 +1,591 @@
+    public_signals: list[Any],
+    verification_key: dict[str, Any],
+) -> None:
+    with tempfile.TemporaryDirectory() as raw_tmp:
+        tmp = pathlib.Path(raw_tmp)
+        proof_path = tmp / "proof.json"
+        public_path = tmp / "public.json"
+        vk_path = tmp / "verification_key.json"
+        proof_path.write_text(json.dumps(proof, sort_keys=True), encoding="utf-8")
+        public_path.write_text(json.dumps(public_signals, sort_keys=True), encoding="utf-8")
+        vk_path.write_text(json.dumps(verification_key, sort_keys=True), encoding="utf-8")
+        result = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)